ion/reader.go

```diff
@@ -148,6 +148,9 @@ type Reader interface {
 	// ByteValue returns the current value as a byte slice (if that makes sense). It returns
 	// an error if the current value is not an Ion clob or an Ion blob.
 	ByteValue() ([]byte, error)
+
+	// InStruct indicates if the reader is currently positioned in a struct.
+	InStruct() bool
```

Comment:
Matter of taste: I'd prefer the name `IsInStruct` to make it obvious that this returns a `bool`.

Reply:
Renamed to ```IsInStruct```